packages/blade/src/components/IconButton/__tests__/IconButton.native.test.tsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+import { fireEvent } from '@testing-library/react-native';
+
+import IconButton from '../';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';
+import { CloseIcon } from '~components/Icons';
+
+describe('<IconButton />', () => {
+  it('should render', () => {
+    const noop = () => {};
+    const { toJSON } = renderWithTheme(
+      <IconButton accessibilityLabel="Close" icon={CloseIcon} onClick={noop} />,
+    );
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should handle callback', () => {
+    const onClick = jest.fn();
+    const { getByRole } = renderWithTheme(
+      <IconButton accessibilityLabel="Close" icon={CloseIcon} onClick={onClick} />,
+    );
+    const iconButton = getByRole('button');
+    fireEvent.press(iconButton);
+
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it('should have a11y label', () => {
```

Comment:
nit: 
```suggestion
  it('should have accessibility label', () => {
```

Reply:
I think `a11y` is more accessible to read and type than `accessibility` 😅